app/src/test/java/org/mozilla/fenix/components/PrivateShortcutCreateManagerTest.kt

```diff
@@ -0,0 +1,107 @@
+package org.mozilla.fenix.components
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.content.IntentSender
+import androidx.core.content.pm.ShortcutInfoCompat
+import androidx.core.content.pm.ShortcutManagerCompat
+import androidx.core.graphics.drawable.IconCompat
+import io.mockk.every
+import io.mockk.mockkClass
+import io.mockk.mockkConstructor
+import io.mockk.mockkStatic
+import io.mockk.spyk
+import io.mockk.verify
+import org.junit.Test
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.home.intent.StartSearchIntentProcessor
+import java.util.UUID
+
+
+class PrivateShortcutCreateManagerTest {
+
+    @Test
+    fun `when createPrivateShortcut is called and Pin shortcut is supported then requestPinShortcut should be called`(){
+        val context = mockkClass(Context::class)
+
+        val privateShortcutCreateManager = spyk(PrivateShortcutCreateManager)
+
+        mockkConstructor(Intent::class)
+
+        val homeActivityIntent = getMockedHomeActivityIntent()
+
+        val shortcut = getMockedShortcutInfoCompat(context, homeActivityIntent)
+
+        val intentSender = getMockedIntentSender(context)
+
+        mockShortcutCompatManagerAndMethods(context, shortcut, intentSender)
+
+        privateShortcutCreateManager.createPrivateShortcut(context)
+
+        verify (exactly = 1) { ShortcutManagerCompat.isRequestPinShortcutSupported(context) }
+        verify (exactly = 1) { ShortcutManagerCompat.requestPinShortcut(context,shortcut,intentSender) }
```

Comment:
I'd say ideally we'd want to capture the `shortcut` and `intentSender` as constructed from the SUT and then check for all the properties we expect to be there.

Reply:
Ok for this one way I could think of is to separate out is `getting intent sender, homeActivity intent` and `shortcut` into different functions and then `use assertions`
WDYT ?

Reply:
If you are breaking `createPrivateShortcut` in multiple methods tested individually here you'd basically test them again.
So maybe you could have some 
> fun `assert IntentSender is correctly built`(actualIntent: PendingIntent) { .. }
etc

methods containing all the assertions for one particular component and then call these both when you're testing each individual method and when testing them all in `requestPinShortcut` to avoid duplicating that checks.

Or you could not refactor `createPrivateShortcut` and then you wouldn't have duplicated checks. Just once, splitted in different assert methods or not.

Reply:
Let's say If I don't refractor, then How could I capture the internal object creation ? 😅